jwst/white_light/white_light.py

```diff
@@ -31,57 +31,94 @@ def white_light(input_model, min_wave=None, max_wave=None):
 
     # The input should contain one row per integration for each spectral
     # order.  NIRISS SOSS data can contain up to three orders.
-    norders = 0  # number of different spectral orders
-    sporders = []  # list of spectral order numbers
-    ntables_order = []  # number of tables for each spectral order
-    prev_spectral_order = -999
-    for i in range(ntables):
-        # The following assumes that all rows for a given spectral order
-        # are contiguous.
-        spectral_order = input_model.spec[i].spectral_order
-        if spectral_order is None:
-            norders = 1
-            sporders = [0]
-            ntables_order = [ntables]
-            break
-        if spectral_order != prev_spectral_order:
+    n_spec = len(input_model.spec)
+    sporders = []  # list of spectral orders available
+    order_list = np.empty((n_spec,))  # order for each spectrum, length nspectra
+    mid_times = np.empty((n_spec,))
+    fluxsums = np.empty((n_spec,))
+
+    # Loop over the spectra in the input model and find mid times and fluxes
+    problems = 0
+    for i, spec in enumerate(input_model.spec):
+        # Figure out the spectral order for this spectrum
+        spectral_order = getattr(spec, "spectral_order", None)
+        if spectral_order not in sporders:
             sporders.append(spectral_order)
-            prev_spectral_order = spectral_order
-            ntables_order.append(1)
-            norders = len(sporders)
+        order_list[i] = spectral_order
+
+        # Figure out mid times for this spectrum
+        mid_time = getattr(spec, "mid_time_mjd", None)
+        if mid_time is None:
+            problems += 1
+            mid_times[i] = np.nan
+            continue
         else:
-            ntables_order[norders - 1] += 1
-
-    log.debug(
-        "norders = %d, sporders = %s, ntables_order = %s",
-        norders,
-        str(sporders),
-        str(ntables_order),
-    )
-
-    # Create a wavelength mask, using cutoffs if specified, then
-    # compute the flux sum for each integration in the input.
-    low_cutoff = -1.0
-    high_cutoff = 1.0e10
-    if min_wave is not None:
-        low_cutoff = min_wave
-    if max_wave is not None:
-        high_cutoff = max_wave
-
-    for i in range(ntables):
+            mid_times[i] = mid_time
+
+        # Create a wavelength mask, using cutoffs if specified, then
+        # compute the flux sum for each integration in the input.
         wave_mask = np.where(
-            (input_model.spec[i].spec_table["WAVELENGTH"] >= low_cutoff)
-            & (input_model.spec[i].spec_table["WAVELENGTH"] <= high_cutoff)
+            (spec.spec_table["WAVELENGTH"] >= min_wave)
+            & (spec.spec_table["WAVELENGTH"] <= max_wave)
         )[0]
+        fluxsums[i] = np.nansum(spec.spec_table["FLUX"][wave_mask])
+
+    if problems > 0:
+        log.warning(
+            f"There were {problems} spectra with no mid time ({100.0 * problems / n_spec}). "
```

Comment:
This message is a little confusing as is.
```suggestion
            f"There were {problems} spectra with no mid time ({100.0 * problems / n_spec}% of total). "
```

Reply:
Thanks for catching this, I think I just forgot to finish writing it

Reply:
let me know if https://github.com/spacetelescope/jwst/pull/9431/commits/cd9b66c2b26d5814c7b0e9fa814083355b80d7bc  addresses this appropriately